setup.py

```diff
@@ -41,9 +41,9 @@ def create_extension() -> setuptools.Extension:
 
     extra_compile_args = {
         "windows": ["/DANTLR4CPP_STATIC", "/Zc:__cplusplus"],
-        "linux": ["-std=c++11"],
-        "darwin": ["-std=c++11"],
-        "cygwin": ["-std=c++11"],
+        "linux": ["--std=c++17"],
+        "darwin": ["--std=c++17", "-mmacosx-version-min=10.14"],
+        "cygwin": ["--std=c++17"],
```

Comment:
Keep the single dash
```suggestion
        "linux": ["-std=c++17"],
        "darwin": ["-std=c++17", "-mmacosx-version-min=10.14"],
        "cygwin": ["-std=c++17"],
```